app/controllers/admin/departments_controller.rb

```diff
@@ -67,6 +74,40 @@ def set_department
       @department = Department.find(params[:id])
     end
 
+    # allow_admin_only
+    # ==================================================
+    # Name : David J. Davis
+    # Date : 2/10/2017
+    #
+    # Description:
+    # Users the admin controller to check if the user is an admin.
+    # if not a flash message is added to the UI and the user is re-directed.
+
+    def allow_admin_only
+      if !check_is_admin
+        redirect_to libraries_url, error: 'You do not have admin access to create or delete libraries.'
```

Comment:
Try thoughts on drying out this by adding it to the admin_controller with parametrs for the url and message.